frontend/public/components/utils/dropdown.jsx

```diff
@@ -283,9 +283,22 @@ class Dropdown_ extends DropdownMixin {
   }
 
   applyTextFilter_(autocompleteText, items) {
-    const { autocompleteFilter } = this.props;
+    const { autocompleteFilter, weightedSort } = this.props;
     if (autocompleteFilter && !_.isEmpty(autocompleteText)) {
-      items = _.pickBy(items, (item, key) => autocompleteFilter(autocompleteText, item, key));
+      if (weightedSort) {
+        // Use fuzzysort
+        let totalItems = _.map(items, (item) => item.props);
+        const result = autocompleteFilter(autocompleteText, totalItems);
+        totalItems = {};
+        _.forEach(result, (item) => {
+          const key = `${item.obj.name}-${item.obj.kind}`;
+          totalItems[key] = items[key];
+        });
+        items = totalItems;
```

Comment:
I would use `Array.reduce` here.

```suggestion
        const totalItems = _.map(items, (item) => item.props);
        const result = autocompleteFilter(autocompleteText, totalItems);
        items = result.reduce((item, acc) => {
          const key = `${item.obj.name}-${item.obj.kind}`;
          acc[key] = items[key];
          return acc;
        }, {});

```